mypyc/irbuild/function.py

```diff
@@ -857,3 +858,44 @@ def gen_dispatch_func_ir(
     func_decl = FuncDecl(dispatch_name, None, builder.module_name, sig)
     dispatch_func_ir = FuncIR(func_decl, args, blocks)
     return dispatch_func_ir
+
+
+def sort_with_subclasses_first(
```

Comment:
This seems plausible, but I think that it might be more straightforward to implement this as a topological sort, instead. I'll follow up via PM.